cli/lsp/text.rs

```diff
@@ -73,10 +73,10 @@ pub fn get_edits(a: &str, b: &str, line_index: &LineIndex) -> Vec<TextEdit> {
   if a == b {
     return vec![];
   }
-  // Heuristic to detect things like minified files. `diff()` is expensive.
-  if b.chars().filter(|c| *c == '\n').count()
-    > line_index.inner.utf8_offsets_len() * 3
-  {
+  // Heuristic to detect things like large JSON or minified files. `diff()` is
+  // expensive.
+  let b_lines = b.chars().filter(|c| *c == '\n').count();
+  if b_lines > 10000 || b_lines > line_index.inner.utf8_offsets_len() * 3 {
```

Comment:
Wouldn't it be better to obtain this value from runtime? Maybe "best guessing" is not fair for all cases 🤔

Reply:
I consider this a heuristic to determine what sort of file the target is, rather than a threshold based on system resources. It's simpler that way

Reply:
This can be optimised further by not scanning the entire file for newlines. It is enough to `.take(10001.max(line_index.inner.utf8_offsets_len() * 3))` of them. That way, we only have to scan 10 % of a file with 100K lines.